src/Analyzers/Core/Analyzers/RemoveUnnecessarySuppressions/SuppressMessageAttributeState.cs

```diff
@@ -118,21 +128,16 @@ public bool HasInvalidOrMissingTarget(ImmutableArray<(string name, IOperation va
                 return targetSymbolString != null;
             }
 
-            var resolvedSymbols = s_listPool.Allocate();
-            try
-            {
-                var resolver = new TargetSymbolResolver(_compilation, targetScope, targetSymbolString);
-                resolvedSymbols.Clear();
-                resolver.Resolve(resolvedSymbols);
-                return resolvedSymbols.Count == 0;
-            }
-            finally
-            {
-                s_listPool.Free(resolvedSymbols);
-            }
+            var resolver = new TargetSymbolResolver(_compilation, targetScope, targetSymbolString);
+            resolvedSymbols = resolver.Resolve(out targetHasDocCommentIdFormat);
+            return resolvedSymbols.IsEmpty;
         }
 
-        private static bool TryGetNamedArgument(ImmutableArray<(string name, IOperation value)> namedAttributeArguments, string argumentName, out string? argumentValue)
+        private static bool TryGetNamedArgument(
+            ImmutableArray<(string name, IOperation value)> namedAttributeArguments,
```

Comment:
slightly confused why these aren't IArguments instead of IOperations

Reply:
This is due to lack of rich IOperation interface for attributes: https://github.com/dotnet/roslyn/issues/18198. See current IOperation tree shape for named attribute arguments: https://github.com/dotnet/roslyn/blob/725e3fc446ec9b07e92cda11b5903ea0c884f232/src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IArgument.cs#L3272-L3295